crates/bevy_pbr/src/light.rs

```diff
@@ -480,14 +481,61 @@ fn cluster_to_index(cluster_dimensions: UVec3, cluster: UVec3) -> usize {
     ((cluster.y * cluster_dimensions.x + cluster.x) * cluster_dimensions.z + cluster.z) as usize
 }
 
+// Sort point lights with shadows enabled first, then by a stable key so that the index
+// can be used to render at most `MAX_POINT_LIGHT_SHADOW_MAPS` point light shadows, and
+// we keep a stable set of lights visible
+pub(crate) fn point_light_order(
+    (entity_1, shadows_enabled_1): (&Entity, &bool),
+    (entity_2, shadows_enabled_2): (&Entity, &bool),
+) -> std::cmp::Ordering {
+    shadows_enabled_1
+        .cmp(shadows_enabled_2)
+        .reverse()
+        .then_with(|| entity_1.cmp(entity_2))
+}
+
 // NOTE: Run this before update_point_light_frusta!
 pub fn assign_lights_to_clusters(
     mut commands: Commands,
     mut global_lights: ResMut<VisiblePointLights>,
     mut views: Query<(Entity, &GlobalTransform, &Camera, &Frustum, &mut Clusters)>,
     lights: Query<(Entity, &GlobalTransform, &PointLight)>,
 ) {
-    let light_count = lights.iter().count();
+    let mut lights = lights.iter().collect::<Vec<_>>();
```

Comment:
I think only the translation is used from the transform. would it make sense to only keep that before collecting?

Reply:
we use everything - note that this lights variable shadows the query (otherwise it'd be better to remove it from the query).

Reply:
I mean something like 
```suggestion
    let mut lights = lights.iter().map(|(e, gt, pl)| (e, gt.translation, pl)).collect::<Vec<_>>();
```
scale and rotation from the lights are not used. I'm not sure of the perf impact, but it would change the footprint from `Vec3 + Vec3 + Quat` to just `Vec3`

Reply:
sorry, you're quite right. i've done that now.